real_preprocessing/src/corner_detection.cpp

```diff
@@ -0,0 +1,90 @@
+#include "ros/ros.h"
+#include "apriltag_ros/AprilTagDetectionArray.h"
+#include "tf_conversions/tf_eigen.h"
+#include "Eigen/Dense"
+#include "fstream"
+#include "string"
+
+void april_detection(const apriltag_ros::AprilTagDetectionArray::ConstPtr& msg)
```

Comment:
Function names should be camelCased i.e. this should be aprilDetection. See [here](http://wiki.ros.org/CppStyleGuide) for the ROS C++ styleguide. Instructions for using clang-format are [here](https://github.com/davetcoleman/roscpp_code_format).